### PR TITLE
Bug fixes for get_spectra in other configs

### PR DESCRIPTION
**Pull request by rosteen**
This fixes two bugs:

- `cubeviz.specviz.get_spectra` now properly retrieves the collapsed spectra rather than the whole cubes.
- `mosviz.specviz.get_spectra` no longer hits an `AttributeError` if there's a spatial subset defined. 

In regards to the second one, I think that is actually a `get_data` bug, since it's returning a Glue Data objects even with `cls=Spectrum1D` set in the internal `get_data` call. A better fix would be to either filter out spatial subset here in `get_spectra` if the config is Mosviz, or to fix the underlying behavior in `get_data`. But this simple fix was the fastest thing to get in - if there are objections to this, I'll look into more later this afternoon.

### CODE REVIEW

**Comment by pllim** [comment]
> fix the underlying behavior in get_data

I think we should fix before releasing it to the public. Otherwise it is going to come back and haunt us.

**Comment by rosteen** [comment]
@pllim We decided to remove the ability to make spatial subsets in the image viewer of Mosviz altogether, but I think that deserves its own PR. These `Spectrum1D` checks can stay IMO, although I could see an argument that they might hide another bug at some point and it would be better to just see an error.

File: jdaviz/configs/specviz/helper.py
Comment (pllim) [inline jdaviz/configs/specviz/helper.py:98]: I say we remove this workaround. Otherwise we'll forget it is there and it would make future debugging confusing.

File: jdaviz/configs/specviz/helper.py
Comment (kecnry) [inline jdaviz/configs/specviz/helper.py:81]: I wonder if the original implementation would have worked with just `viewer.state` instead of `viewer` (to be a bit more general in case a profile-viewer with collapse-function is ever added to a different/custom config?

Comment (rosteen) [reply jdaviz/configs/specviz/helper.py:81]: It turns out the spectrum viewers still have a not-`None` default `function` in the other configs, I tried.

File: jdaviz/configs/specviz/helper.py
Comment (rosteen) [reply jdaviz/configs/specviz/helper.py]: I removed the check, but still request `cls=Spectrum1D` from `get_data`.

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

Might need a rebase to get CI to pass.

**Comment by rosteen** [comment]
I'm tempted to merge with one review, but I didn't mark this trivial...

**Comment by duytnguyendtn** [comment]
Reviewing now! I can merge for you once I approve if you're alright with that?

**Comment by duytnguyendtn** [comment]
(Or you could wait for me to approve I guess 😅)

**Review by duytnguyendtn — APPROVED** [review]
Confirmed in Cubeviz the spatial subset does not trigger the `get_spectra` traceback, as well as obeys the statistic in Plot options. Would be nice to have some tests for get_spectra obeying the statistic drop down, but I'm willing to let that slide for the release. Thanks for the quick fixes!

**Comment by rosteen** [comment]
Thanks for the speedy reviews!

**Pull request merged**
